testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -738,30 +739,32 @@ void testCastToExactNumeric(CastType castType, SqlOperatorFixture f) {
   @MethodSource("safeParameters")
   void testCastStringToDecimal(CastType castType, SqlOperatorFixture f) {
     f.setFor(SqlStdOperatorTable.CAST, VmName.EXPAND);
-    if (!DECIMAL) {
-      return;
-    }
     // string to decimal
     f.checkScalarExact("cast('1.29' as decimal(2,1))",
         "DECIMAL(2, 1) NOT NULL",
-        "1.3");
+        "1.2");
```

Comment:
Different systems have different interception methods, I think the return value `1.3` is also reasonable. Most systems return `1.3`, such as MySQL, PG, Hive, and Spark, but ClickHouse returns `1.2`.

Reply:
There doesn't seem to be a standard set of sql rules here, so maybe we should think about what makes sense mathematically or what makes sense to follow the rules of most databases

Reply:
@YiwenWu I have written about the rounding mode used in Calcite: https://issues.apache.org/jira/browse/CALCITE-6322. Here I am just consistent with the rest of the codebase, but I agree that I would prefer a different rounding mode. Maybe this should be part of the typesystem.